.github/copilot-instructions.md

```diff
@@ -0,0 +1,262 @@
+# Roulette Web Application
+A Flutter web application for decision-making through a digital roulette wheel. Users can add items to the roulette, spin the wheel, and share their lists via URL. Built with Flutter, using BLoC state management and deployed to GitHub Pages.
+
+Always reference these instructions first and fallback to search or bash commands only when you encounter unexpected information that does not match the info here.
+
+## Working Effectively
+
+### Prerequisites and Setup
+- Install Flutter SDK 3.8.0 or higher: `https://docs.flutter.dev/get-started/install`
+- Ensure you have Chrome or another web browser for testing
+- For development, Docker with `ghcr.io/cirruslabs/flutter:stable` image can be used
+
+### Bootstrap, Build, and Test Commands
+```bash
+# Install dependencies - takes 30-60 seconds
+flutter pub get
+
+# **NOTE**: Code generation files are already committed to the repository
+# If you need to regenerate (only after modifying @freezed or @JsonSerializable files):
+# dart run build_runner build --delete-conflicting-outputs
+# This command may fail in some environments - the generated files are already present
+
+# NEVER CANCEL: Build for web release - takes 3-5 minutes
+# Set timeout to 10+ minutes
+# May fail due to network restrictions downloading Flutter Web SDK
+flutter build web --release --base-href /roulette/
+
+# NEVER CANCEL: Build for web debug (faster, no network dependencies) - takes 2-3 minutes
+# Set timeout to 8+ minutes
+# Recommended approach if network access is limited  
+flutter build web --debug
+
+# NEVER CANCEL: Run tests - takes 1-2 minutes
+# Set timeout to 5+ minutes
+# May fail due to dependency resolution issues in containerized environments
+flutter test
+
+# Run analysis (linting) - takes 10-30 seconds
+# Currently shows errors due to dependency resolution, but still useful for checking syntax
+flutter analyze --no-fatal-infos
```

Comment:
Shouldn't show errors about dependency resolution after running flutter pub get. Check this, if this is the case there are two possible causes:
- If chaining them fixes it, we probably don't have any issues. Then we need to just create a named container at the start of the work, and run docker exec on it every time instead of creating a new container every time
- if the issues are not fixed by chaining, we need to create an issue for fixing this.